src/main/cpp/src/hash.cuh

```diff
@@ -134,4 +134,18 @@ std::unique_ptr<cudf::column> xxhash64(
   rmm::cuda_stream_view stream      = cudf::get_default_stream(),
   rmm::device_async_resource_ref mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Computes the Hive hash value of each row in the input set of columns.
+ *
+ * @param input The table of columns to hash
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ *
+ * @returns A column where each row is the hash of a column from the input.
+ */
+std::unique_ptr<cudf::column> hive_hash(
+  cudf::table_view const& input,
+  rmm::cuda_stream_view stream      = cudf::get_default_stream(),
+  rmm::device_async_resource_ref mr = rmm::mr::get_current_device_resource());
```

Comment:
There is an issue with function declarations here. These declarations are put in a `cuh` file, then included in `*Jni.cpp` files which are not compiled under `nvcc` but regular `g++`. Thus, in the future we may have compiling issue when `g++` decides to throw compile error when seeing (unused) device code in `.cuh` file.

The solution now should be extracting these function declarations into a separate `hash.hpp` file.

Reply:
done

Reply:
This is a great catch.  I did notice that we were putting this in a `.cuh`.  But I didn't realize that this was then included in a `.cpp`.